qiskit/result/counts.py

```diff
@@ -54,38 +54,43 @@ def __init__(self, data, time_taken=None, creg_sizes=None,
                 ``[('c_reg', 2), ('my_creg', 4)]``.
             memory_slots (int): The number of total ``memory_slots`` in the
                 experiment.
+            include_zeros (bool): If `True`, it includes the counts in zero. Default `False`.
+
         Raises:
             TypeError: If the input key type is not an int or string
             QiskitError: If a dit string key is input with creg_sizes and/or
                 memory_slots
         """
-        bin_data = None
         data = dict(data)
-        if not data:
-            self.int_raw = {}
-            self.hex_raw = {}
-            bin_data = {}
-        else:
+
+        self.int_raw = {}
+        self.hex_raw = {}
+        bin_data = {}
+
+        if include_zeros:
+            for slot_int in range(memory_slots):
```

Comment:
You should have a check that memory_slots is actually defined here. If you do `range(None)` this will fail.

Reply:
Fixed in 25fbd24